### PR TITLE
AGENT-374: agent-based-client: Apply installconfig overrides

### PR DESCRIPTION
**Pull request by celebdor**
The annotation was mistakenly ignored. This sets it on the cluster.

It's recommended to take a few extra minutes to provide more information about
how this code was tested. Here are some questions that may be worth answering:

- Should this PR be tested by the reviewer? no
- Is this PR relying on CI for an e2e test run? yes
- Should this PR be tested in a specific environment? agent based installation
- Any logs, screenshots, etc that can help with the review process?

## List all the issues related to this PR

- [ ] New Feature
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] Agent based installation
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed
- [x] No tests performed yet

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: cmd/agentbasedinstaller/register.go [hunk 1]
```diff
@@ -75,7 +75,21 @@ func RegisterCluster(ctx context.Context, log *log.Logger, bmInventory *client.A
 		return nil, errorutil.GetAssistedError(registerClusterErr)
 	}
 
-	log.Info("Registered cluster with id: " + clusterResult.Payload.ID.String())
+	log.Infof("Registered cluster with id: %s", clusterResult.Payload.ID)
+
+	annotations := aci.GetAnnotations()
+	if installConfigOverrides, ok := annotations["agent-install.openshift.io/install-config-overrides"]; ok {
```
[the remark this answers is not in the document]
Comment (celebdor) [reply cmd/agentbasedinstaller/register.go:81]: Thanks!

**Comment by omertuc** [comment]
Unrelated, but why is this an annotation and not a proper CRD spec field? It makes sense when we do it on BMHs because we don't own that CRD, but we do own ACI, so why annotation?

**Comment by celebdor** [comment]
> Unrelated, but why is this an annotation and not a proper CRD spec field? It makes sense when we do it on BMHs because we don't own the BMH CRD, but we do own ACI, so why annotation?

I was thinking the same. We should add the field in a future release

**Comment by celebdor** [comment]
I had forgotten to git add the aci yaml :P

File: cmd/agentbasedinstaller/register.go [hunk 2]
```diff
@@ -75,7 +75,21 @@ func RegisterCluster(ctx context.Context, log *log.Logger, bmInventory *client.A
...
-	log.Info("Registered cluster with id: " + clusterResult.Payload.ID.String())
+	log.Infof("Registered cluster with id: %s", clusterResult.Payload.ID)
+
+	annotations := aci.GetAnnotations()
+	if installConfigOverrides, ok := annotations[controllers.InstallConfigOverrides]; ok {
+		updateInstallConfigParams := &installer.V2UpdateClusterInstallConfigParams{
+			ClusterID:           *clusterResult.Payload.ID,
+			InstallConfigParams: installConfigOverrides,
+		}
+		_, updateClusterErr := bmInventory.Installer.V2UpdateClusterInstallConfig(ctx, updateInstallConfigParams)
```
Comment (zaneb) [inline cmd/agentbasedinstaller/register.go:86]: I suspect that ignoring the return value here is the reason the new test is failing.

**Pull request merged**